cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -0,0 +1,1205 @@
+        THRUST_NS_QUALIFIER::unwrap_contiguous_iterator(::cuda::std::get<Is>(in)), ublkcp_alignment)...);
+  }
+#endif // _CUB_HAS_TRANSFORM_UBLKCP
+
+  template <typename ActivePolicy, std::size_t... Is>
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE cudaError_t
+  invoke_algorithm(cuda::std::index_sequence<Is...>, ::cuda::std::integral_constant<Algorithm, Algorithm::memcpy_async>)
+  {
+    auto ret = configure_memcpy_async_kernel<ActivePolicy>();
+    if (!ret)
+    {
+      return ret.error;
+    }
+    // TODO(bgruber): use a structured binding in C++17
+    // auto [launcher, kernel, elem_per_thread] = *ret;
+    return ::cuda::std::get<0>(*ret).doit(
+      ::cuda::std::get<1>(*ret),
+      num_items,
+      ::cuda::std::get<2>(*ret),
+      op,
+      out,
+      make_aligned_base_ptr_kernel_arg<THRUST_NS_QUALIFIER::try_unwrap_contiguous_iterator_t<RandomAccessIteratorsIn>>(
+        THRUST_NS_QUALIFIER::unwrap_contiguous_iterator(::cuda::std::get<Is>(in)), memcpy_async_alignment)...);
+  }
+
+  template <typename ActivePolicy, std::size_t... Is>
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE cudaError_t invoke_algorithm(
+    cuda::std::index_sequence<Is...>, ::cuda::std::integral_constant<Algorithm, Algorithm::unrolled_staged>)
+  {
+    using policy_t      = typename ActivePolicy::algo_policy;
+    const auto grid_dim = static_cast<unsigned int>(
+      ::cuda::ceil_div(num_items, Offset{policy_t::BLOCK_THREADS * policy_t::ITEMS_PER_THREAD}));
+    return THRUST_NS_QUALIFIER::cuda_cub::launcher::triple_chevron(grid_dim, policy_t::BLOCK_THREADS, 0, stream)
+      .doit(kernel,
+            num_items,
+            /* items per thread taken from policy */ -1,
+            op,
+            out,
+            make_iterator_kernel_arg<THRUST_NS_QUALIFIER::try_unwrap_contiguous_iterator_t<RandomAccessIteratorsIn>>(
+              THRUST_NS_QUALIFIER::try_unwrap_contiguous_iterator(::cuda::std::get<Is>(in)))...);
+  }
+
+  template <typename ActivePolicy, std::size_t... Is>
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE cudaError_t
+  invoke_algorithm(cuda::std::index_sequence<Is...>, ::cuda::std::integral_constant<Algorithm, Algorithm::prefetch>)
+  {
+    using policy_t          = typename ActivePolicy::algo_policy;
+    constexpr int block_dim = policy_t::BLOCK_THREADS;
+    int max_occupancy       = 0;
+    const auto error        = MaxSmOccupancy(max_occupancy, kernel, block_dim, 0);
+    if (error != cudaSuccess)
+    {
+      return error;
+    }
+
+    const int items_per_thread =
+      loaded_bytes_per_iter == 0
+        ? +policy_t::ITEMS_PER_THREAD_NO_INPUT
+        : ::cuda::ceil_div(ActivePolicy::min_bif, max_occupancy * block_dim * loaded_bytes_per_iter);
+    const int items_per_thread_clamped =
+      ::cuda::std::clamp(items_per_thread, +policy_t::MIN_ITEMS_PER_THREAD, +policy_t::MAX_ITEMS_PER_THREAD);
+    const Offset tile_size = block_dim * items_per_thread_clamped;
```

Comment:
Can you elaborate why? The type of `block_dim * items_per_thread_clamped` is `int`. `Offset` can only be `int32_t` (which should be `int`) or `int64_t`.

Reply:
My concern is that compilers may warn about implicit casts here. Since there were other discussions on this PR about "what if `OffsetT` was unsigned", I reviewed the PR with the mental state that the implementation could, in future, also have an `unsigned` type for `OffsetT`. Feel free to discard `unsigned`/`signed` `OffsetT` comments if that's not a concern for us.

Reply:
Ok, then I would further argue: If compilers warn about an implicit cast here, we should find out with the unit tests and CI.

The discussion on the allowed offset types is not finished yet, so we may want to support unsigned types eventually. But then we should add those to the unit tests, at which time we will also see conversion warnings.

At the moment, offsets can get negative in the memcpy_async and ublkcp kernel, but this can be worked around if necessary.